cli/internal/lockfile/pnpm_lockfile.go

```diff
@@ -0,0 +1,234 @@
+// Reference https://github.com/pnpm/pnpm/blob/main/packages/lockfile-types/src/index.ts
+type PnpmLockfile struct {
+	Version            float32                    `yaml:"lockfileVersion"`
+	Importers          map[string]ProjectSnapshot `yaml:"importers"`
+	Packages           map[string]PackageSnapshot `yaml:"packages,omitempty"`
+	NeverBuiltDeps     []string                   `yaml:"neverBuiltDependencies,omitempty"`
+	OnlyBuiltDeps      []string                   `yaml:"onlyBuiltDependencies,omitempty"`
+	Overrides          map[string]string          `yaml:"overrides,omitempty"`
+	PackageExtChecksum string                     `yaml:"packageExtensionsChecksum,omitempty"`
+	PatchedDeps        map[string]PatchFile       `yaml:"patchedDependencies,omitempty"`
+}
+
+var _ Lockfile = (*PnpmLockfile)(nil)
+
+// ProjectSnapshot Snapshot used to represent projects in the importers section
+type ProjectSnapshot struct {
+	Specifiers           map[string]string         `yaml:"specifiers"`
+	Dependencies         map[string]string         `yaml:"dependencies,omitempty"`
+	OptionalDependencies map[string]string         `yaml:"optionalDependencies,omitempty"`
+	DevDependencies      map[string]string         `yaml:"devDependencies,omitempty"`
+	DependenciesMeta     map[string]DependencyMeta `yaml:"dependenciesMeta,omitempty"`
+	PublishDirectory     string                    `yaml:"publishDirectory,omitempty"`
+}
+
+// PackageSnapshot Snapshot used to represent a package in the packages setion
+type PackageSnapshot struct {
+	ID string `yaml:"id,omitempty"`
+
+	Resolution PackageResolution `yaml:"resolution,flow"`
+	Engines    struct {
+		Node string `yaml:"node"`
+		NPM  string `yaml:"npm,omitempty"`
+	} `yaml:"engines,omitempty,flow"`
+	CPU           []string `yaml:"cpu,omitempty,flow"`
+	Os            []string `yaml:"os,omitempty,flow"`
+	HasBin        bool     `yaml:"hasBin,omitempty"`
+	RequiresBuild bool     `yaml:"requiresBuild,omitempty"`
+
+	PeerDependencies     map[string]string `yaml:"peerDependencies,omitempty"`
+	PeerDependenciesMeta map[string]struct {
+		Optional bool `yaml:"optional"`
+	} `yaml:"peerDependenciesMeta,omitempty"`
+	Dependencies         map[string]string `yaml:"dependencies,omitempty"`
+	OptionalDependencies map[string]string `yaml:"optionalDependencies,omitempty"`
+	TransitivePeerDeps   []string          `yaml:"transitivePeerDependencies,omitempty"`
+	BundledDependencies  []string          `yaml:"bundledDependencies,omitempty"`
+
+	Dev      bool `yaml:"dev"`
+	Optional bool `yaml:"optional,omitempty"`
+	Patched  bool `yaml:"patched,omitempty"`
+	Prepare  bool `yaml:"prepare,omitempty"`
+
+	// only needed for packages that aren't in npm
+	Name    string `yaml:"name,omitempty"`
+	Version string `yaml:"version,omitempty"`
+
+	LibC       []string `yaml:"libc,omitempty"`
+	Deprecated string   `yaml:"deprecated,omitempty"`
+}
+
+// PackageResolution Various resolution strategies for packages
+type PackageResolution struct {
```

Comment:
There appears to be an optional `registry` field here.

https://github.com/pnpm/pnpm/blob/e6ee26e9c7bcc58b22721a182970cf029a20203a/packages/lockfile-types/src/index.ts#L40